Packs/TeamViewer/Integrations/TeamViewerEventCollector/TeamViewerEventCollector.py

```diff
@@ -0,0 +1,223 @@
+from CommonServerUserPython import *  # noqa
+from datetime import datetime
+import urllib3
+from typing import Any, Dict, Tuple, List, Optional
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+
+''' CONSTANTS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'  # ISO8601 format with UTC, default in XSOAR
+DEFAULT_LIMIT = 300
+VENDOR = 'teamviewer'
+PRODUCT = 'tensor'
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    Client class to interact with the service API
+    """
+
+    def __init__(self, base_url, verify, proxy, headers):
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy, headers=headers)
+
+    def get_events(self, params=None, body=None):
+        return self._http_request(
+            method='POST', url_suffix='/api/v1/EventLogging',
+            headers=self._headers, params=params, data=body)
+
+
+''' HELPER FUNCTIONS '''
+
+
+def search_events(client: Client, limit: int,
+                  body: Optional[Dict[str, Any]] = None
+                  ) -> Tuple[List[Dict[str, Any]], CommandResults]:
+    """
+    Searches for T alerts.
+    Args:
+        client: Client, client to use.
+        limit: int, the limit of the results to return.
+        body: dict, contains the time parameters.
+    Returns:
+        list: A list containing the events
+    """
+    results: List[Dict] = []
+    token_next_page = None
+    next_page = True
+    if limit <= 0:
+        raise DemistoException('the limit argument cannot be negative or zero.')
+    while next_page:
+        response = client.get_events(body=body)
+        demisto.debug(f'http response:\n {response}')
+        results += response.get('AuditEvents', [])
+        next_page = response.get('ContinuationToken')
+        if token_next_page := response.get('ContinuationToken'):
+            if not body:
+                body = {}
+            body['ContinuationToken'] = token_next_page
+        else:
+            next_page = False
+            demisto.debug('finished fetching http response')
+    events: List[Dict[str, Any]] = sorted(results, key=lambda x: x['Timestamp'])
+    if limit < len(events):
+        events = events[-limit:]
+    hr = tableToMarkdown(name='Events', t=events) if events else 'No events found.'
+    return events, CommandResults(readable_output=hr)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client, first_fetch_time: datetime) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.get_events(body={'StartDate': first_fetch_time.strftime(DATE_FORMAT),
```

Comment:
Why you don't pass here limit:1?
We don't need anymore or the whole default limit in the test command

Reply:
added api command to verify the token